src/jit/morph.cpp

```diff
@@ -14051,11 +14051,21 @@ GenTree* Compiler::fgMorphModToSubMulDiv(GenTreeOp* tree)
     {
         numerator = fgMakeMultiUse(&tree->gtOp1);
     }
+    else if (lvaLocalVarRefCounted && numerator->OperIsLocal())
+    {
+        // Morphing may introduce new lclVar references. Increase ref counts
+        lvaRecursiveIncRefCounts(numerator);
+    }
 
     if (!denominator->OperIsLeaf())
     {
         denominator = fgMakeMultiUse(&tree->gtOp2);
```

Comment:
And FWIW re: the ref counts in the non-leaf case: `fgMakeMultiUse` increments the ref counts for the created lclVar three times: once for the assign, once for the first use, and once for the new lclVar node that it returns. This is correct for the purposes of this transform.